tensorboard/plugins/debugger/tf_debugger_dashboard/tf-source-code-view.html

```diff
@@ -0,0 +1,673 @@
+          lineElement.classList.add('highlighted-source-line');
+          // Expand the node name list.
+          self._toggleLineNodes(highlightLineno, true);
+        }
+        self.set('_highlightedElements', highlightedElements);
+
+        this._highlightNodeElements(nodeName);
+      });
+    },
+
+    // Highlight all node elements corresponding to the specified node.
+    // A node may correspond to multiple node elements, because the creation
+    // stack of the node may involve more than one line in the current source
+    // file.
+    // Args:
+    //   nodeName: Name of the node, without base expansion.
+    _highlightNodeElements(nodeName) {
+      // Remove the higlighting on the node elements of the pervious focus node.
+      if (this._oldFocusNodeName !=  null) {
+        for (const nodeElement of this._nodeName2NodeElements[this._oldFocusNodeName]) {
+          nodeElement.style['font-weight'] = 'normal';
+        }
+      }
+      // Add highlighting on the current focus node.
+      for (const nodeElement of this._nodeName2NodeElements[nodeName]) {
+        nodeElement.style['font-weight'] = 'bold';
+      }
+      // Scroll to the node element at the topmost stack position in the
+      // current source file, unless this is a continue-to action originated from
+      // a given node element, which is not necessarily the topmost stack position.
+      if (this._setHightlightOriginNodeElement == null) {
+        this._nodeName2StackTopNodeElement[nodeName].scrollIntoView(
+            {block: 'center', behaviour: 'smooth'});
+      } else {
+        this.set('_setHightlightOriginNodeElement', null);
+      }
+      this.set('_oldFocusNodeName', nodeName);
+    },
+
+    // Populate the display of node full stack.
+    // Args:
+    //   nodeName: The name of the node whose stack is to be shown.
+    //   currentFullFilePath: The full path to the source file currently being
+    //     displayed.
+    //   curerntLineno: The current source line number being displayed.
+    _populateFullStack(nodeName, currentFullFilePath, currentLineno) {
+      this._querySourceCodeEndPoint(
+          {'mode': 'op_traceback', 'op_name': nodeName}).then(response => {
+        const stackContent = this.$$('#full-stack-content');
+        // First, remove all children of the full stack content div.
+        while (stackContent.firstChild) {
+          stackContent.removeChild(stackContent.firstChild);
+        }
+        // Then, create all the clickable links.
+        for (const stackFrame of response['op_traceback'][nodeName]) {
+          const frameElement = document.createElement('li');
+          const fullFilePath = stackFrame[0];
+          const lineno = Number(stackFrame[1]);
+          frameElement.textContent = fullFilePath + ': ' + String(lineno);
+          if (_.contains(this._fullFilePaths, fullFilePath)) {
+            frameElement.classList.add('stack-frame-clickable');
+            // TODO(cais): Make class CSS work.
```

Comment:
See other comment.

Reply:
See other reply. Unfortunately adding Polymer.dom() still doesn't work for me. I hope we can look into it together sometime this wee.